opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -229,17 +249,24 @@ namespace Opm
                         // perfPhaseRates
                         const int oldPerf_idx_beg = (*it).second[ 1 ];
                         const int num_perf_old_well = (*it).second[ 2 ];
+                        const auto new_iter = this->wellMap().find(well.name());
+                        if (new_iter == this->wellMap().end()) {
+                            throw std::logic_error {
+                                well.name() + " is not in internal well map - "
+                                "Bug in WellStateFullyImplicitBlackoil"
+                            };
+                        }
+
+                        const int connpos = new_iter->second[1];
+                        const int num_perf_this_well = new_iter->second[2];
```

Comment:
What do the number of connections in the wellMap reflect?
Are they aware of parallelism (i.e. that the number connection on the local grid might differ from the number of connections in a serial run? If not, then this will cause problems for distributed wells.

Reply:
> What do the number of connections in the `wellMap` reflect?

Neither more nor less than what's [implied](https://github.com/OPM/opm-simulators/blob/009feaa35df69aa10902388b082b2f42bf0b010d/opm/simulators/wells/WellState.hpp#L84-L92) by the [`well_perf_data_`](https://github.com/OPM/opm-simulators/blob/009feaa35df69aa10902388b082b2f42bf0b010d/opm/simulators/wells/BlackoilWellModel_impl.hpp#L734-L736).  This PR just switches to using the value we already calculated rather than having to rederive it from `well_perf_data_[w].size()`.

> Are they aware of parallelism (i.e. that the number connection on the local grid might differ from the number of connections in a serial run?

No idea.  Unless you took special care to handle that when you introduced the distributed wells then probably not.

Reply:
Thanks, I checked it. According to [`BlackoilWellModel::initializeWellPerfData`](/OPM/opm-simulators/blob/33e3102983287345c62c1158f839351add02fab5/opm/simulators/wells/BlackoilWellModel_impl.hpp#L637-L669) this should be ok.